tests/jballerina-unit-test/src/test/resources/test-src/isolation-analysis/isolation_inference_with_variables_3.bal

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2023 WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+int[] a = [];
+
+int[][] b = [[1, 2], a]; // a is not an isolated expression here
+
+int[][] c = [[1, 2], a.clone()];
```

Comment:
why is the RHS side an isolated expression?

Reply:
A list constructor is an isolated expression if all of its members are isolated expressions (https://ballerina.io/spec/lang/master/#isolated_expressions).
- `[1, 2]` - `1` and `2` are isolated expressions because they are subtypes of `readonly`
- `a.clone()` is isolated because "A function-call-expr is isolated if the type of the function being called is isolated and the expression for every argument is isolated. In addition, a call to the clone or cloneReadOnly functions defined by the lang.value module of the lang library is always isolated." (https://ballerina.io/spec/lang/master/#function_call)